### DISCUSSION TITLE
Connecting from ESP8266 to EMQX on local docker

### DISCUSSION THREAD

**Discussion by Trojan13** [emqx/emqx#8413] [Q&A] [answered]
Hello, I am using EMQX for a small project focused on edge computing (thats why I need a local broker).
For this I have started a local docker with EMQX on a laptop. Additionally I have an ESP8266 connected to a sensor and I want to send the data via MQTT to the EMQX broker. For this I use the library "PubSubClient.h". I have already tried the MQTT connection from my PC with MQTTX and it works without any problems. So it can't be due to a firewall. 
To further troubleshoot, I tried the official snippet on the ESP8266 to connect to the test broker [(Snippet-Link)](https://docs.emqx.com/en/cloud/latest/connect_to_deployments/esp8266.html#connection) .This also works. But as soon as I change the address (mqtt_broker ) with my laptops local IP + "/mqtt" and the port to 8084, no connection can be established. Is there anything else I should consider if I want to use a local emqx broker?

**Comment by zmstone** [comment, marked as answer]
Hi 
Just wan to make sure you are using the right port:
8084 is the wss:// (secure websocket) listener’s port by default.
 
 mqtts:// (MQTT over TLS) is port 8883.

**Reply by Trojan13** [reply]
Thanks for your fast reply.  I tried every single combination of ports I can think of.
I think the ESP has a problem resolving the name adress with the added "/mqtt". 
Is there any way to get rid of the "/mqtt"-Path in a local EMQX broker?

Here is a log:
```
The client ESP8266Client-f237 connects to the public mqtt broker
[hostByName] request IP for: 192.168.178.53/mqtt
[hostByName] Host: 192.168.178.53/mqtt IP: 192.168.178.255
failed with state -2
```

**Reply by zmstone** [reply]
I’m not sure why you need to try combinations of port numbers.
If it’s mqtt over tls, then pick 8883
If it’s mqtt over websocket over TLS pick 8084
If it’s mqtt over tcp pick 1883
If it’s mqtt over websocket pick 8083

The /mqtt suffix to the host string only makes sense for websocket client sdk? I don’t know what it is sorry

**Reply by Trojan13** [reply]
Alright I feel a bit stupid now but atleast it works.
This was the important part:
"The /mqtt suffix to the host string only makes sense for websocket client sdk?"

I am now using the local ip, port 1883 and no host string and thats what works. I always got state -2 or -1 from PubSubClient and tried so many things. But this fixed it. If anyone is interested in the .ino feel free to ask. 

Thank you so much! :)

**Comment by joshemq** [comment]
Hi @Trojan13, would you be open to sharing more details about your project with me? I'm EMQ's marketing director in the US, and I'd appreciate the chance to interview you and maybe share your project with others if you'd be interested. Would it be okay to contact you directly?
